glassfish-on-suse/azuredeploy.json

```diff
@@ -0,0 +1,539 @@
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of storage account"
+      }
+    },
+    "publicIPAddressName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Public IP Name"
+      }
+    },
+    "dnsNameforLBIP": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "DNS for Load Balancer IP"
+      }
+    },
+    "vnetName": {
+      "type": "string",
+      "defaultValue": "vnet",
+      "metadata": {
+        "description": "VNET name"
+      }
+    },
+    "lbName": {
+      "type": "string",
+      "defaultValue": "lb",
+      "metadata": {
+        "description": "Load Balancer name"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_DS2_v2",
+      "allowedValues": [
+        "Standard_A4",
+        "Standard_A5",
+        "Standard_A6",
+        "Standard_A7",
+        "Standard_A8",
+        "Standard_A9",
+        "Standard_A10",
+        "Standard_A11",
+        "Standard_D2",
+        "Standard_D3",
+        "Standard_D4",
+        "Standard_D11",
+        "Standard_D12",
+        "Standard_D13",
+        "Standard_D14",
+        "Standard_D2_v2",
+        "Standard_D3_v2",
+        "Standard_D4_v2",
+        "Standard_D5_v2",
+        "Standard_D11_v2",
+        "Standard_D12_v2",
+        "Standard_D13_v2",
+        "Standard_D14_v2",
+        "Standard_G1",
+        "Standard_G2",
+        "Standard_G3",
+        "Standard_G4",
+        "Standard_G5",
+        "Standard_DS2",
+        "Standard_DS3",
+        "Standard_DS4",
+        "Standard_DS11",
+        "Standard_DS12",
+        "Standard_DS13",
+        "Standard_DS14",
+        "Standard_DS2_v2",
+        "Standard_DS3_v2",
+        "Standard_DS4_v2",
+        "Standard_DS5_v2",
+        "Standard_DS11_v2",
+        "Standard_DS12_v2",
+        "Standard_DS13_v2",
+        "Standard_DS14_v2",
+        "Standard_GS1",
+        "Standard_GS2",
+        "Standard_GS3",
+        "Standard_GS4",
+        "Standard_GS5"
+      ],
+      "metadata": {
+        "description": "VM(s) size"
+      }
+    },
+    "osImage": {
+      "type": "string",
+      "allowedValues": [ "opensuse", "sles" ],
+      "defaultValue": "opensuse",
+      "metadata": {
+        "description": "Operating System to use for VM image(s)"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username"
+      }
+    },
+    "glassfishAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "GlassFish admin user password"
+      }
+    },
+    "sshPublicKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "SSH public key on all VMs"
+      }
+    }
+  },
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "apiVersionStorage": "2015-06-15",
```

Comment:
Do you mean I should avoid listing apiVersions as variables? It seems much more flexible to only list the values once.